src/js/libcs/OpImageDrawing.js

```diff
@@ -385,3 +385,169 @@ evaluator.cameravideo$0 = function() {
+
+    if (!img) return nada;
+
+    var w = img.width;
+    var h = img.height;
+
+    var w0 = evaluateAndHomog(args[0]);
+    var w1 = evaluateAndHomog(args[1]);
+    var v0 = evaluateAndHomog(List.realVector([0, h, 1]));
+    var v1 = evaluateAndHomog(List.realVector([w, h, 1]));
+
+    if (w0 === nada || w1 === nada || p === nada) return nada;
+
+    //create an orientation-reversing similarity transformation that maps w0->v0, w1->v1
+    var ii = List.ii;
+    var jj = List.jj;
+
+    var m1 = eval_helper.basismap(v0, v1, ii, jj); //interchange I and J,
+    var m2 = eval_helper.basismap(w0, w1, jj, ii); //see Thm. 18.4 of Perspectives on Projective Geometry
+    var p = evaluateAndHomog(args[3]);
+    var coord = eval_helper.extractPoint(General.mult(m1, General.mult(List.adjoint3(m2), p)));
+
+    if (!coord.ok) return nada;
+
+    if (interpolate) {
+        coord.x -= 0.5; //center of pixels are in the middle of them.
+        coord.y -= 0.5; //Now pixel-centers have wlog integral coordinates
+    }
+
+    if (repeat) {
+        coord.x = (coord.x % w + w) % w;
+        coord.y = (coord.y % h + h) % h;
+    }
+
+    var rgba = [0, 0, 0, 0];
+
+    if (interpolate) {
+        var i, j;
+
+        var xf = (coord.x % 1 + 1) % 1; //fractional parts
+        var yf = (coord.y % 1 + 1) % 1;
+        var pixels = readPixelsIndirection(img, Math.floor(coord.x), Math.floor(coord.y), 2, 2);
+
+        //modify pixels for boundary cases:
+        if (repeat) { //read pixels at boundary seperately
+            if (Math.floor(coord.x) === w - 1 || Math.floor(coord.y) === h - 1) {
+                var p10 = readPixelsIndirection(img, (Math.floor(coord.x) + 1) % w, Math.floor(coord.y), 1, 1);
+                var p01 = readPixelsIndirection(img, Math.floor(coord.x), (Math.floor(coord.y) + 1) % h, 1, 1);
+                var p11 = readPixelsIndirection(img, (Math.floor(coord.x) + 1) % w, (Math.floor(coord.y) + 1) % h, 1, 1);
+                pixels = pixels.slice(0, 4).concat(p10).concat(p01).concat(p11);
+            }
+        } else { //clamp to boundary
+            if (Math.floor(coord.x) === -1 && xf >= 0.5)
+                for (i = 0; i < 4; i++)
+                    for (j = 0; j < 2; j++) pixels[8 * j + i] = pixels[8 * j + i + 4];
+            if (Math.floor(coord.x) === w - 1 && xf < 0.5)
+                for (i = 0; i < 4; i++)
+                    for (j = 0; j < 2; j++) pixels[8 * j + i + 4] = pixels[8 * j + i];
+            if (Math.floor(coord.y) === -1 && yf >= 0.5)
+                for (i = 0; i < 8; i++) pixels[i] = pixels[i + 8];
+            if (Math.floor(coord.y) === h - 1 && yf < 0.5)
+                for (i = 0; i < 8; i++) pixels[i + 8] = pixels[i];
+        }
+
+        //bilinear interpolation for each component i
+        for (i = 0; i < 4; i++)
+            for (j = 0; j < 4; j++) {
+                rgba[i] += pixels[4 * j + i] * ((j < 2) ? (1 - yf) : yf) * ((j % 2 === 0) ? (1 - xf) : xf);
+            }
```

Comment:
Personally I think it would be easier to write this without a loop, but I don't want to force you if you disagree.

```js
rgba[i] = (1 - yf) * ((1 - xf) * pixels[i] + xf * pixels[i + 4]) +
    yf * ((1 - xf) * pixels[i + 8] + xf * pixels[i + 12]);
```